torch/testing/_internal/distributed/rpc/rpc_test.py

```diff
@@ -1626,3 +1630,19 @@ class TestPickler():
         with _use_rpc_pickler(test_pickler):
             self.assertTrue(torch.distributed.rpc.api._default_pickler is test_pickler)
         self.assertTrue(torch.distributed.rpc.api._default_pickler is _internal_rpc_pickler)
+
+    @dist_init
+    def test_function_not_on_callee(self):
+        # test that if a function does not exist on a callee, we don't crash,
+        # instead we get an AttributeError indicating that the func does not exist.
+        if self.rank != 0:
+            # Use delattr to remove the binding of a func on callee nodes
+            import sys
+            this_module = sys.modules[__name__]
+            delattr(this_module, "foo_add")
```

Comment:
Question: I wonder do we have to explicitly delete the attr? Does it work if we define `foo_add` as a local function within the test?

Reply:
@mrshenli If we define the function locally, then on the caller side (worker0 in this case) the pickler fails with a `Cannot pickle local function` error. So it looks like that case already has error handling.